packages/astro/src/vite-plugin-astro-server/css.ts

```diff
@@ -13,37 +12,48 @@ interface ImportedStyle {
 /** Given a filePath URL, crawl Vite’s module graph to find all style imports. */
 export async function getStylesForURL(
 	filePath: URL,
-	loader: ModuleLoader,
-	mode: RuntimeMode
+	loader: ModuleLoader
 ): Promise<{ urls: Set<string>; styles: ImportedStyle[] }> {
 	const importedCssUrls = new Set<string>();
 	// Map of url to injected style object. Use a `url` key to deduplicate styles
 	const importedStylesMap = new Map<string, ImportedStyle>();
 
 	for await (const importedModule of crawlGraph(loader, viteID(filePath), true)) {
 		if (isBuildableCSSRequest(importedModule.url)) {
-			let ssrModule: Record<string, any>;
-			try {
-				// The SSR module is possibly not loaded. Load it if it's null.
-				ssrModule = importedModule.ssrModule ?? (await loader.import(importedModule.url));
-			} catch {
-				// The module may not be inline-able, e.g. SCSS partials. Skip it as it may already
-				// be inlined into other modules if it happens to be in the graph.
-				continue;
+			// In dev, we inline all styles if possible
+			let css = '';
+			// If this is a plain CSS module, the default export should be a string
+			if (typeof importedModule.ssrModule?.default === 'string') {
+				css = importedModule.ssrModule.default;
 			}
-			if (
-				mode === 'development' && // only inline in development
```

Comment:
I removed this check (and the `mode` param entirely) as it seems like we only use this function in dev now.